### ISSUE TITLE
Explicite default column width

### ISSUE THREAD

**Issue by Codeberg-AsGithubAlternative-buhtz** [DeepSpace2/StyleFrame#131]
Different from `pandas.to_excel()` it seems to me that `StyelFrame.to_excel()` does explicit set the width of each column based on its values. I am not sure here it is just a hypothesis.

Please see the full code at the end of that report.

The excel result looks like this
![image](https://user-images.githubusercontent.com/11861496/176139876-d5ee91a1-af77-43c5-80ef-07d5bc4d3097.png)

I assume that column "B" was set to a default minimum with because there is much space before and after the values.
I also assume that column "C" was set to the with of its longest value (`33123`).
I assume that the with of headers are ignored.

Now I do a "auto fit column width" in Excel (in German "Spaltenbreite automatisch anpassen"):
![image](https://user-images.githubusercontent.com/11861496/176140439-af6b6c70-a5ba-4b62-8e52-2434588bd9ff.png)

The index column `A` is the only that was fit to the with of its values/contents.
Column `B` and `C` does not change. That is why I assume there an explicit width.

The default Excel behavior would be to set the width that it fit to the content/values of a column:
![image](https://user-images.githubusercontent.com/11861496/176141524-7c6ecb87-7e7f-4985-aff1-dfa2d45dc4b8.png)

Even you format that table as a "table with header" in Excel the auto-fit function create this
![image](https://user-images.githubusercontent.com/11861496/176141415-e37d254e-eb92-4cfc-bb8c-aa32564558c5.png)

```
#!/usr/bin/env python3
import os
import sys
import pathlib
import pandas
import styleframe
import styleframe.utils

print(styleframe._versions_)

df = pandas.DataFrame(
    data={
        'idx1': list('AABB'),
        'column with long caption': [1234, 345, 33123, 2],
        }
    )
print(df)

file_path = pathlib.Path.cwd() / 'test.xlsx'

default_style = styleframe.Styler(font_size=14)
sf = styleframe.StyleFrame(df, styler_obj=default_style)

sf.to_excel(file_path, index=True).save()

os.system(str(file_path))

```

Changing this would break the default behaviour of your users. I am aware of this.
But maybe there could be switch or something to avoid the setting of an explicit width.
IMHO better default behaviour would be to auto-fit the width.

**Comment by DeepSpace2** [comment]
Interesting topic.

We do have `best_fit` option for `to_excel`, it just does not take into account the column name, just the column content. That **is** probably an unexpected behavior, at least with how the documentation is currently worded (the note says "...  correct column-width based on the longest value in each provided column"): 

>Parameters
>best_fit (None or [str](https://docs.python.org/3/library/stdtypes.html#str) or [list](https://docs.python.org/3/library/stdtypes.html#list) or [tuple](https://docs.python.org/3/library/stdtypes.html#tuple) or [set](https://docs.python.org/3/library/stdtypes.html#set)) –
>single column, list, set or tuple of columns names to attempt to best fit the width for.
>
>Note
>best_fit will attempt to calculate the correct column-width based on the longest value in each provided column. However this isn’t guaranteed to work for all fonts (works best with monospaced fonts). The formula used to calculate a column’s width is equivalent to
>
>`(len(longest_value_in_column) + A_FACTOR) * P_FACTOR`
>The default values for `A_FACTOR` and `P_FACTOR` are 13 and 1.3 respectively, and can be modified before calling >StyleFrame.to_excel by directly modifying `StyleFrame.A_FACTOR` and `StyleFrame.P_FACTOR`

Consider this:

```python
import styleframe

long_string = 'a' * 100
data_with_long_column_name = {long_string: ['a']}
data_with_long_content = {'a': [long_string]}
long_column_name_sf = styleframe.StyleFrame(data_with_long_column_name)
long_content_sf = styleframe.StyleFrame(data_with_long_content)

long_column_name_sf.to_excel('width_test/long_column_name_sf_without_best_fit.xlsx').save()
long_content_sf.to_excel('width_test/long_content_sf_without_best_fit.xlsx').save()
long_column_name_sf.to_excel('width_test/long_column_name_sf_with_best_fit.xlsx', best_fit=long_string).save()
long_content_sf.to_excel('width_test/long_content_sf_with_best_fit.xlsx', best_fit='a').save()
```

long_column_name_sf_without_best_fit.xlsx:
![image](https://user-images.githubusercontent.com/6841988/178081537-00fea912-c319-4480-93d3-4f2a317b7488.png)


long_content_sf_without_best_fit.xlsx:
![image](https://user-images.githubusercontent.com/6841988/178081488-7c8eb514-76c7-4636-a30b-1ca806f2a79d.png)


long_column_name_sf_with_best_fit.xlsx:
![image](https://user-images.githubusercontent.com/6841988/178081459-466d3b8e-23be-462e-b3cb-5a67690c22d9.png)


long_content_sf_with_best_fit.xlsx:
![image](https://user-images.githubusercontent.com/6841988/178081519-80297508-d8bd-4dc7-8bfd-23718e27feb2.png)

**Comment by Codeberg-AsGithubAlternative-buhtz** [comment]
I can not provide a PR for this because I don't understand the details here. I understand your examples. But I would say that they doesn't matter for my problem.
You mentioned `best_fit`. But I don't use this argument and it`s default is `None`. So IMHO `StyleFrame` shouldn't touch the width options in that case.

**Comment by DeepSpace2** [comment]
> StyleFrame shouldn't touch the width options in that case.

I disagree. Pandas' defaults are pretty useless.  How is this useful in any way? You can't even tell on what column the headers is:

```python
import pandas as pd

pd.DataFrame({'a' * 100: []}).to_excel('output.xlsx')
```

![image](https://user-images.githubusercontent.com/6841988/178157578-a87bbf10-bc05-461a-91c7-a4f0409fafba.png)

If the long content is in a row (ie not in the header) it looks a bit better, but still messy. I don't want StyleFrame's output to look this way, not even by default.

![image](https://user-images.githubusercontent.com/6841988/178157634-b87e6b54-143a-4251-833f-105ebdd83ad5.png)

**Comment by Codeberg-AsGithubAlternative-buhtz** [comment]
I see your point and agree with you.

You say "Pandas' defaults are pretty useless". From the viewpoint of StyleFrame you are right.
But from Pandas viewpoint it is not. There are no "defaults" in the context of formating the Excel output with `DataFrame.to_excel()`. Pandas don't take care of the visual part. This is not a bug or missing feature. It is just by design because Pandas don't want to take care of visual part and formating.

That is where your *StyleFrame* comes in. It is its job.

My initial problem was
 - The "auto fit column width" in Excel (in German "Spaltenbreite automatisch anpassen") doesn't work anymore when I use *StyleFrame*. I assume this is because you manipulate the width of columns. But I don't understand all the details of the technical reasons here.

I understand that *StyleFrame* tries and have to set some defaults for column widths. That is fine.
But that defaults shouldn't "block" Excel default behavior ("auto fit column width").

I don't know how you manipulate the column width. But isn't it possible to manipulate the column width **without** making it a fixed width so that the "auto fit column width" feature doesn't work anymore?
